contrib/scalar/scalar.c

```diff
@@ -5,11 +5,197 @@
 #include "cache.h"
 #include "gettext.h"
 #include "parse-options.h"
+#include "config.h"
+
+static void setup_enlistment_directory(int argc, const char **argv,
+				       const char * const *usagestr,
+				       const struct option *options)
+{
+	if (startup_info->have_repository)
+		BUG("gitdir already set up?!?");
```

Comment:
> Is this necessary? Perhaps replace with `assert(!startup_info->have_repository);` so debug builds will catch it?

I vastly prefer explicit `BUG()` calls over `assert()` calls, as I am totally unaware of anyone running with debug builds where the latter would be anything but a noop.

> Has this already taken @vdye's contributions in [microsoft#402](https://github.com/microsoft/git/pull/402)? Maybe it should, at least in the full non-RFC version.

No, those contributions have not yet been considered. TBH I am unsure what to do about `scalar register` in Git worktrees whose path does not end in `/src`. I was under the impression that that is not even supported, and it sure muddles an otherwise clean concept of what a Scalar enlistment is...

Reply:
The idea of `scalar register` is that we can apply our recommended config and start background maintenance on an existing Git repository that was created by a normal `git clone`. We can't adjust the partial clone or sparse-checkout for such a repo, but `scalar register` still have _some_ value there.